### ISSUE TITLE
Dataset format and Best Practices For Language Model Fine-tuning

### ISSUE THREAD

**Issue by HanGuo97** [huggingface/transformers#1305]
## ❓ Questions & Help

Hi, thanks for making this code base available!

I have two questions, one on the input format of for fine-tuning the language model on custom dataset, and one on (unreasonably-)long data preprocessing time. Thanks in advance for any help!

- I'm trying to fine-tune the BERT Model on an extra dataset, and am using the `run_lm_finetuning.py` script in the `examples/` directory. However, I'm having trouble locating instructions on the proper format of the input data. There used to be some instructions in the `examples/lm_finetuning/` directory, but they seem deprecated now.

- As a start, I followed the `run_lm_finetuning.py` example and changed nothing but `--train_data_file` argument with a bigger text file (arbitrary format). The training, however, hangs on the data preprocessing part for about 10 hours, and the last standard output is shown below.

```
pytorch_transformers.tokenization_utils -   Token indices sequence length is longer than the specified maximum sequence length for this model (164229992 > 512). Running this sequence through the model will result in indexing errors
```

**Comment by gr8Adakron** [comment]
I am facing the same issue as there is no proper format available for defining the train and test dataset. 
As usual, I use .csv file in a format of columns with (UID, Text, and Labels). But according to the Wiki.txt its more of arbitrary format.

Any help would be appreciated.

**Comment by anassalamah** [comment]
I'm having the same issue. I think it's counting the total length of the tokenized corpus not only the tokenized document length. I tried to run the wiki raw files as mentioned in the read me and still get this warning of total tokenized corpus length.

I tried to the following formats with no success:
1. sentence per line with a blank line in between docs
2. document per line with a blank line in between docs
